src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2926,6 +2929,13 @@ internal bool CanEmitBoolean()
             return (diagnostic == null) || (diagnostic.Severity != DiagnosticSeverity.Error);
         }
 
+        internal bool CanEmitString()
+        {
+            var stringType = GetSpecialType(SpecialType.System_String);
+            var diag = stringType.GetUseSiteDiagnostic();
+            return (diag == null) || (diag.Severity != DiagnosticSeverity.Error);
+        }
```

Comment:
Consider extracting a `CanEmitSpecialType(SpecialType type)` helper method, used by `CanEmitBoolean` and `CanEmitString`.